cloudnet/src/main/java/de/dytanic/cloudnet/console/JLine3Console.java

```diff
@@ -0,0 +1,439 @@
+                animation.addToCursor(1);
+            }
+        }
+
+        return this;
+    }
+
+    @Override
+    public boolean hasColorSupport() {
+        return true;
+    }
+
+    @Override
+    public boolean isUsingMatchingHistoryComplete() {
+        return this.matchingHistorySearch;
+    }
+
+    @Override
+    public void setUsingMatchingHistoryComplete(boolean matchingHistoryComplete) {
+        this.matchingHistorySearch = matchingHistoryComplete;
+    }
+
+    @Override
+    public void resetPrompt() {
+        this.prompt = System.getProperty("cloudnet.console.prompt", "&c%user%&r@&7%screen% &f=> &r");
+        this.updatePrompt();
+    }
+
+    @Override
+    public void removePrompt() {
+        this.prompt = null;
+        this.updatePrompt();
+    }
+
+    @Override
+    public void emptyPrompt() {
+        this.prompt = ConsoleColor.DEFAULT.toString();
+        this.updatePrompt();
+    }
+
+    @Override
+    public void clearScreen() {
+        this.terminal.puts(InfoCmp.Capability.clear_screen);
+        this.terminal.flush();
+    }
+
+    @Override
+    public void close() throws Exception {
+        this.animationThreadPool.shutdownNow();
+        this.consoleReadThread.interrupt();
+
+        this.terminal.flush();
+        this.terminal.close();
+
+        AnsiConsole.systemUninstall();
+    }
+
+    @Override
+    public @NotNull String getPrompt() {
+        return this.prompt;
+    }
+
+    @Override
+    public void setPrompt(@NotNull String prompt) {
+        this.prompt = prompt;
+        this.updatePrompt();
+    }
+
+    @Override
+    public @NotNull String getScreenName() {
+        return this.screenName;
+    }
+
+    @Override
+    public void setScreenName(@NotNull String screenName) {
+        this.screenName = screenName;
+    }
+
+    private void updatePrompt() {
+        this.prompt = ConsoleColor.toColouredString('&', this.prompt)
+                .replace("%version%", VERSION)
+                .replace("%user%", USER)
+                .replace("%screen%", this.screenName);
+        this.lineReader.setPrompt(this.prompt);
+    }
+
+    private void print(@NotNull String text) {
+        this.lineReader.getTerminal().puts(InfoCmp.Capability.carriage_return);
+        this.lineReader.getTerminal().puts(InfoCmp.Capability.clr_eol);
+        this.lineReader.getTerminal().writer().print(text);
+        this.lineReader.getTerminal().writer().flush();
+
+        this.redisplay();
+    }
+
+    private void redisplay() {
+        if (!this.lineReader.isReading()) {
+            return;
+        }
+
+        this.lineReader.callWidget(LineReader.REDRAW_LINE);
+        this.lineReader.callWidget(LineReader.REDISPLAY);
+    }
+
+    private void toggleHandlers(boolean enabled, @NotNull Collection<?> handlers) {
+        for (Object handler : handlers) {
+            ((ConsoleHandler<?>) handler).setEnabled(enabled);
+        }
+    }
+
+    @NotNull
```

Comment:
Codacy found an issue: [Use explicit scoping instead of the default package private level](https://app.codacy.com/gh/CloudNetService/CloudNet-v3/pullRequest?prid=5923591)